### PR TITLE
add error handler support

### LINKED ISSUE #43: Error Handlers not supported
Hi there,

### Terraform Version

0.12.9
### Affected Resource(s)

rundeck_job

### Issue

I have read the documentation to create a workflow for the job and noticed error handling support has been missing at command level.

### Expected Behavior
What should have happened?

### Actual Behavior
What actually happened?
Feature doesn't exist.

### PR DESCRIPTION
**Pull request by kiuby**
fix #43 

Any help for adding tests is more than welcome :)
(it's working with a running instance of rundeck)

### CODE REVIEW

File: website/docs/r/job.html.md [hunk 1]
```diff
@@ -186,6 +188,14 @@ A command's `step_plugin` or `node_step_plugin` block both have the following st
 
 * `config`: (Optional) Map of arbitrary configuration parameters for the selected plugin.
 
+A command's `errorhandler` block both have the following structure:
+
+* `type`: (Required) The name of the plugin to execute.
+
+* `config`: (Optional) Map of arbitrary configuration parameters for the selected plugin.
+
+* `keepgoingonsuccess`: (Optional) A boolean to configure if the error handler job should be launched on the success of the job.
```
Comment (vijayanandweinv) [inline website/docs/r/job.html.md:197]: I think some conventions are missing here.

`keepgoingonsuccess` should be `keep_going_on_success`

File: website/docs/r/job.html.md [hunk 2]
```diff
@@ -186,6 +188,14 @@ A command's `step_plugin` or `node_step_plugin` block both have the following st
 
 * `config`: (Optional) Map of arbitrary configuration parameters for the selected plugin.
 
+A command's `errorhandler` block both have the following structure:
```
Comment (vijayanandweinv) [inline website/docs/r/job.html.md:191]: `errorhandler` should be `error_handler`

File: rundeck/resource_job.go [hunk 3]
```diff
@@ -791,6 +835,16 @@ func jobToResourceData(job *JobDetail, d *schema.ResourceData) error {
 				}
 			}
 
+			if command.ErrorHandler != nil && command.ErrorHandler.NodeStepPlugin != nil {
+				commandConfigI["errorhandler"] = []interface{}{
```
Comment (vijayanandweinv) [inline rundeck/resource_job.go:839]: same convention of `snakecase` should be followed here in all attributes

**Comment by kiuby** [comment]
@vijayanandweinv everything has been fixed :) thanks for your feedback !

**Comment by kiuby** [comment]
@ProTip  if you can add this PR into the next milestone it will be great !
(And if you need help feel free to reach me, we've got some hacks to have a working version of this provider, it will be great to remove it :) )

File: rundeck/resource_job.go [hunk 4]
```diff
@@ -321,13 +321,37 @@ func resourceRundeckJob() *schema.Resource {
...
+func resourceRundeckJobCommandResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"keep_going_on_success": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (ProTip) [inline rundeck/resource_job.go:345]: Type is not actually a required property for the error handler. In the case of a simple exec command step as the handler it looks like this in the xml export:
```
<errorhandler keepgoingOnSuccess='true'>
    <exec>true</exec>
</errorhandler>
```

File: website/docs/r/job.html.md [hunk 5]
```diff
@@ -186,6 +188,14 @@ A command's `step_plugin` or `node_step_plugin` block both have the following st
 
 * `config`: (Optional) Map of arbitrary configuration parameters for the selected plugin.
 
+A command's `error_handler` block both have the following structure:
+
+* `type`: (Required) The name of the plugin to execute.
```
Comment (ProTip) [inline website/docs/r/job.html.md:193]: Error handler can be any step type? I don't believe a step plugin is required..

File: rundeck/resource_job.go [hunk 6]
```diff
@@ -321,13 +321,37 @@ func resourceRundeckJob() *schema.Resource {
...
+			},
+			"config": {
```
Comment (ProTip) [inline rundeck/resource_job.go:347]: This should probably be a typed schema. It matches very closely to the command schema, though the differences may make it tricky to compose in a DRY fashion via the SDK and go..

**Comment by fdevans** [comment]
@kiuby any updates based on @ProTip feedback?

**Comment by fdevans** [comment]
Closing due to no follow-up on feedback from our team.

**Pull request closed without merging**